Updating slack post custom message step to support multiple channels.

marcusdroy opened the pull request:
This adds support for #3 to post to multiple slack channels for a payload.

marcusdroy committed:
Updating slack post custom message step to support multiple channels.

nhmathis commented on plugin/plugin.xml:
```diff
@@ -69,8 +69,8 @@
-      <property name="channel" required="true">
-        <property-ui description="Your public channel within Slack where UrbanCode Deploy messages will be sent. The WebHook must have access to this channel." label="Channel" type="textBox"/>
+      <property name="channels" required="true">
+        <property-ui description="Your public channels within Slack where UrbanCode Deploy messages will be sent. The WebHook must have access to this channels." label="Channels" type="textAreaBox"/>
```
Please indicate how you expect the channels to be specified.

nhmathis replied:
From my understanding, the channels should be separated by new lines.

nhmathis commented on src/main/scripts/postToSlackCustom.groovy:
```diff
@@ -21,35 +21,50 @@ final def props = airTool.getStepProperties()
 // properties
 final def webhook = props['webhook'];
-final def slackChannel = props['channel'];
+final def slackChannels = props['channels'];
```
You'll want to migrate the channel property to the channels property so no data is lost during upgrade.

You need to add a quick migration tag regardless in the upgrade.xml, but it should look like this...

`<migrate to-version ="4">`
` <migrate-command name="Post Custom Notification to Slack">`
`<migrate-properties>`
`<migrate-property name="channels" old="channel" default="" />`
`</migrate-properties>`
`</migrate-command>`
`</migrate>`

nhmathis reviewed (changes requested):
Please also add a quick release note to the info.xml and README.md files.

nhmathis commented on src/main/scripts/postToSlackCustom.groovy:
```diff
@@ -21,35 +21,50 @@ final def props = airTool.getStepProperties()
+   }
+
+   if (!slackChannel.startsWith("@") && !slackChannel.startsWith("#")) {
+      return
```
Quick println saying what you are skipping is helpful for debug. 

println "Warning::  The channel '${slackChannel}' is not formatted correctly. It must begin with an @ or # symbol."

marcusdroy committed:
Adding in aditional error checks for post to slack message.

nhmathis commented:
Hi, 

Please make updates to the upgrade.xml and plugin.xml as indicated above.

- Nick

Commit:
Xml updates and Fail early

- Updated plugin info and upgrade xml to migrate and update channels
properly
- Fail immediately if any of the channels are invalid

Commit:
Updated to ensure each slackChannel is used
